packages/@sanity/form-builder/src/inputs/files/common/FileInputButton/FileInputButton.tsx

```diff
@@ -67,19 +42,47 @@ export const FileInputButton = React.forwardRef(function FileInputButton(
     [onSelect]
   )
 
+  const content = (
+    <Flex align="center" justify="center" padding={padding}>
+      {/* Icon */}
+      {icon && (
+        <Box marginRight={text ? space : undefined}>
+          <Text size={fontSize}>
+            {isValidElement(icon) && icon}
+            {isValidElementType(icon) && createElement(icon)}
+          </Text>
+        </Box>
+      )}
+
+      {/* Text */}
+      {text && (
+        <Text
+          size={fontSize}
+          align={textAlign}
+          textOverflow="ellipsis"
+          weight={theme.sanity.button.textWeight}
+        >
+          {text}
+        </Text>
+      )}
+    </Flex>
+  )
+
   return (
-    <SelectButton {...rest} forwardedAs="label" inputId={id}>
-      <Input
-        type="file"
-        value=""
-        id={id}
-        onChange={handleChange}
-        ref={forwardedRef}
+    <FileButton {...rest} id={id} padding={0}>
+      {content}
+
+      {/* Visibly hidden input */}
+      <input
         accept={accept}
         capture={capture}
+        id={id}
```

Comment:
I was confused because it was called `inputId` before, and this was not intentional. It could be changed to `htmlFor`, but removing works the same in this case so will do that.

Reply:
(PS. It’s still unclear to me what this ID is used for.)

Reply:
> PS. It’s still unclear to me what this ID is used for

Yeah, the inputId that we previously had didn't make any sense, not sure what happened there. Just removing it sounds good to me, but we should probably forward it to the `<input type="file"`? That way you'll be able to create a label `for` it without having to wrap it.

Reply:
Thanks for background!

> we should probably forward the given id prop to the <input type="file"? That way you'll be able to create a label for it without having to wrap it.

That’s what were’re doing now, so I’ll merge this :-)